### PR TITLE
[SymfonyStyle] [Console] Add progressRedrawFrequency method

### PR DESCRIPTION
**Pull request by vchabot**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets |  |
| License | MIT |
| Doc PR | https://github.com/symfony/symfony-docs/pull/6217 |

Hello, 

I made this PR to be able to use the `setRedrawFrequency` method from the `ProgressBar` helper into `SymfonyStyle` class. I also added some tests for the progress bar feature into SymfonyStyle class.

### CODE REVIEW

**Comment by javiereguiluz** [comment]
Just asking: could we make the "redraw frequency" an option instead of a new method?

Before:

```php
public function progressStart($max = 0);
public function progressAdvance($step = 1);
public function progressRedrawFrequency($step = 1);
```

After:

```php
public function progressStart($max = 0, $redrawFrequency = 1);
public function progressAdvance($step = 1);
```

**Comment by javiereguiluz** [comment]
@vchabot the test suite is currently broken. Just ignore errors and focus only on the errors (if any) reported for the files related to your change.

File: src/Symfony/Component/Console/Style/StyleInterface.php [hunk 1]
```diff
@@ -153,6 +153,13 @@ public function progressStart($max = 0);
     public function progressAdvance($step = 1);
 
     /**
+     * Sets the progress redraw frequency.
+     *
+     * @param int $step The frequency in steps
+     */
+    public function progressRedrawFrequency($step = 1);
```
Comment (xabbuh) [inline src/Symfony/Component/Console/Style/StyleInterface.php]: Adding a method to an interface is a BC break.

**Comment by xabbuh** [comment]
Status: Needs work

**Comment by xabbuh** [comment]
@vchabot Yes, just remove it from the interface.

**Comment by javiereguiluz** [comment]
@xabbuh what do you think about adding a method argument instead of a new method?

**Comment by ogizanagi** [comment]
Hi @vchabot . Just asking: Won't using the `ProgressBar` instance returned by `SymfonyStyle::createProgressBar` be sufficient ?
Tweaking the progress bar can imply a lot of method calls, and I'm not sure why we should expose those methods in the `SymfonyStyle` class where using the `ProgressBar` instance would be more efficient.

What do you think ?

**Comment by ogizanagi** [comment]
I don't see any difference between using `SymfonyStyle::createProgressBar` and opening `SymfonyStyle::getProgressBar()`, as both will return the `ProgressBar` instance, properly typehinted. So you won't need to import anything else more :confused:

**Comment by j0k3r** [comment]
@ogizanagi yeah we can use `createProgressBar` to avoid using the `use` statement. The point here is : do we want to keep SymfonyStyle dead simple as a shortcut to other helpers **without** any customization on them or not?

Switching `getProgressBar` to public will allow to customize the ProgressBar internal variable of SymfonyStyle without having to rewrite all the stuff of ProgressBar in the command (start, progress, redraw, finish, etc ..). We can keep using SymfonyStyle.

**Comment by fabpot** [comment]
The `SymfonyStyle` implements defaults for a consistent UI. So, by design, nothing should be configurable. I think this is a won't fix.

**Pull request closed without merging**